lib/open_project/translations/models/locales_updater.rb

```diff
@@ -0,0 +1,127 @@
+require 'pathname'
+require 'fileutils'
+
+require_relative '../helpers/tmp_directory'
+require_relative './git_repository'
+require_relative './i18n_provider'
+require_relative './locales_updater_configuration'
+
+ENGLISH_TRANSLATION_FILE = 'en.yml'
+ENGLISH_JS_TRANSLATION_FILE = 'js-en.yml'
+ACCEPTANCE_LEVEL = ENV['ACCEPTANCE_LEVEL'].nil? ? 100: ENV['ACCEPTANCE_LEVEL'].to_i
+
+class LocalesUpdater
+  include TmpDirectory
+
+  def update_all_locales_of_all_repos(debug: true)
+    repos_to_update = plugins_with_locales
+
+    within_tmp_directory(delete_if_exists: true, debug: debug) do
+      repos_to_update.each do |plugin_name, specifics|
+        update_i18n_handle(specifics)
+
+        within_tmp_directory(path: File.join(FileUtils.pwd, plugin_name), debug: debug) do
+          within_plugin_repo(configuration_hash: specifics, path: FileUtils.pwd, debug: debug) do
+            puts "Uploading english for #{plugin_name}"
+            upload_english
+            request_build
+            puts "Downloading translations for #{plugin_name}"
+            download_and_replace_locales
+          end
+        end
+      end
+    end
+  end
+
+  private
+
+  def within_plugin_repo(configuration_hash:, path:, debug:)
+    setup_plugin_repo(configuration_hash, path)
+    @plugin_repo.within_repo do
+      yield
+    end
+    commit_and_push_plugin_repo(debug)
+  end
+
+  def plugins_with_locales
+    configuration[:plugins]
+  end
+
+  def configuration
+    LocalesUpdaterConfiguration.configuration
+  end
+
+  def update_i18n_handle(configuration_hash)
+    @i18n_provider = begin
+      project_id = configuration_hash[:project_id]
+      api_key = configuration_hash[:api_key]
+      version = configuration_hash[:version]
+      I18nProvider.new(project_id, api_key, version)
+    end
+  end
+
+  def setup_plugin_repo(configuration_hash, path)
+    uri = configuration_hash[:uri]
+    branch = configuration_hash[:branch]
+    previous_branch = configuration_hash[:previous_branch]
+
+    @plugin_repo = GitRepository.new(uri, path)
+    @plugin_repo.clone
+
+    @plugin_repo.checkout(branch)
+    @plugin_repo.merge(previous_branch, strategy: :ours) if previous_branch
+    @plugin_repo
+  end
+
+  def commit_and_push_plugin_repo(debug)
+    @plugin_repo.add('config/locales')
+    @plugin_repo.commit('update locales from crowdin')
+    @plugin_repo.push unless debug
+  end
+
+  def upload_english
+    # either add or update the english (js) translation file
+    titles = {
+      ENGLISH_TRANSLATION_FILE => 'OpenProject Wording',
```

Comment:
@jonasheinrich can you put in the name of the plugin here? I think it would read nicer.